Spigot-API-Patches/0288-add-RespawnFlags-to-PlayerRespawnEvent.patch

```diff
@@ -0,0 +1,73 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Thu, 22 Apr 2021 17:17:54 -0700
+Subject: [PATCH] add RespawnFlags to PlayerRespawnEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java b/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java
+index d2be2ad2e3665728e614a89dd62ef9237f1d3ce6..0fd95edda148333d09b7f57b9738201433dd360f 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java
+@@ -14,17 +14,30 @@ public class PlayerRespawnEvent extends PlayerEvent {
+     private Location respawnLocation;
+     private final boolean isBedSpawn;
+     private final boolean isAnchorSpawn;
++    private final java.util.Set<RespawnFlag> respawnFlags; // Paper
+ 
+     @Deprecated
+     public PlayerRespawnEvent(@NotNull final Player respawnPlayer, @NotNull final Location respawnLocation, final boolean isBedSpawn) {
+         this(respawnPlayer, respawnLocation, isBedSpawn, false);
+     }
+ 
++    @Deprecated // Paper
+     public PlayerRespawnEvent(@NotNull final Player respawnPlayer, @NotNull final Location respawnLocation, final boolean isBedSpawn, final boolean isAnchorSpawn) {
++        // Paper start
++        this(respawnPlayer, respawnLocation, isBedSpawn, isAnchorSpawn, new org.bukkit.event.player.PlayerRespawnEvent.RespawnFlag[]{});
++    }
++
++    public PlayerRespawnEvent(@NotNull final Player respawnPlayer, @NotNull final Location respawnLocation, final boolean isBedSpawn, final boolean isAnchorSpawn, @NotNull final RespawnFlag...respawnFlags) {
```

Comment:
`Set<RespawnFlag>` instead - using varargs in an event constructor is a bad idea.

Reply:
Done